doc/source/analysis.rst

```diff
@@ -1,5 +1,400 @@
 Graph analysis
 ==============
+|igraph| enables analysis of graphs/networks from simple operations such as adding and removing nodes to complex theoretical constructs such as community detection. Read the `API documentation`_ for details on each function and class.
 
-.. note:: TODO. This is a placeholder section; it is not written yet.
+The context for the following examples will be to import the :class:`Graph` class and to have one or more graphs available:
 
+>>> import igraph as ig
+>>> from igraph import Graph
+>>> g = Graph(edges=[[0, 1], [2, 3]])
+
+To get a summary representation of the graph, use :meth:`Graph.summary`. For instance
+
+>>> g.summary(verbosity=1)
+
+will provide a fairly detailed description.
+
+To copy a graph, use :meth:`Graph.copy`.
+
+Vertices and edges
++++++++++++++++++++++++++++
+Vertices are numbered 0 to `n-1`, where n is the number of vertices in the graph. These are called the "vertex ids".
+To count vertices, use :meth:`Graph.vcount`:
+
+>>> n = g.vcount()
+
+Edges also have ids from 0 to `m-1` and are counted by :meth:`Graph.ecount`:
+
+>>> m = g.ecount()
+
+To get a sequence of vertices, use their ids and :attr:`Graph.vs`:
+
+>>> for v in g.vs:
+>>>     print(v)
+
+Similarly for edges, use :attr:`Graph.es`:
+
+>>> for e in g.es:
+>>>     print(e)
+
+.. note:: These two attributes are special sequences with their own useful methods. Some are described below, and all can be found in the `API documentation`_.
+
+If you prefer a vanilla edge list, you can use :meth:`Graph.get_edge_list`.
+
+Incidence
+++++++++++++++++++++++++++++++
+To get the vertices at the two ends of an edge, use :attr:`Edge.source` and :attr:`Edge.target`:
+
+>>> e = g.es[0]
+>>> v1, v2 = e.source, e.target
+
+Vice versa, to get the edge if from the source and target vertices, you can use :meth:`Graph.get_eid` or, for multiple pairs of source/targets,
+:meth:`Graph.get_eids`. The boolean version, asking whether two vertices are directly connected, is :meth:`Graph.are_connected`.
+
+To get the edges incident on a vertex, you can use :meth:`Vertex.incident`, :meth:`Vertex.out_edges` and
+:meth:`Vertex.in_edges`. The three are equivalent on undirected graphs but not directed ones, of course:
+
+>>> v = g.vs[0]
+>>> edges = v.incident()
+
+The :meth:`Graph.adjecent` and :meth:`Graph.incident` functions fulfill the same purpose with a slightly different syntax based on vertex ids:
```

Comment:
`adjacent`, not `adjecent`. But `adjacent` is deprecated (it is not the correct term for this operation), so it should not be mentioned in any tutorials.

Reply:
One other thing that we should agree on is the capitalization of `ids` (or `IDs`). I'm fine with either, I'm not a native English speaker so I can't tell the difference - just let us be consistent.

Reply:
yep, removed in favour of `incident`. I use `id` throughout now, since we have functions such as `get_eid`